src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -577,7 +577,9 @@ public String resolve(String name) {
     public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
 
         public FormValidation doCheckProjectName(
-                @AncestorInPath Job<?,?> anc, @QueryParameter String value) {
+                @AncestorInPath AbstractProject anc, @QueryParameter String value) {
```

Comment:
Why do you need have it `AbstractProject`?
Pipeline (aka. workflow)'s snippet generator requires `Job`.
f28c55475678492a5e0659cb444b062fa5fe748b